src/subcommands/user/index.ts

```
@@ -13,48 +14,81 @@ const logIn = createLeaf({
     password,
   },
   async action(_, opts) {
-    const credentials = credentialsStore.readIfExists();
-    if (credentials) {
-      throw new TerseError(`Already logged in as ${chalk.bold(credentials.email)}`);
+    //
+    try {
+      const username = await checkLoggedIn();
+      if (username) {
+        throw new CodedError(`Already logged in as ${chalk.bold(username)}`, 'LOGGED_IN');
+      }
+    } catch (err) {
+      if (err.code === 'LOGGED_IN') {
+        throw new TerseError(err);
+      }
```

It's discouraged to throw an error in a `try` block that is immediately caught in a `catch` because there's almost always a cleaner way to handle the control flow. Also in this case if the `checkLoggedIn()` call fails for any reason other than one with code LOGGED_IN (which is presumably never) the error is silently eaten because the `catch` block only `throw` on that one error code.

---

This method isn't going to work either @carnesen : Variable 'username' is used before being assigned.

I know what you're saying though... this seems redundant and inefficient to do it this way....